automatic/virtualbox-guest-additions-guest.install/tools/chocolateyInstall.ps1

```diff
@@ -1,45 +1,46 @@
+$binX86Filename = 'VBoxWindowsAdditions-x86.exe'
+$binX64Filename = 'VBoxWindowsAdditions-amd64.exe'
 
-$i = 0
-do {
-    $tempPath = Join-Path -Path $env:TEMP -ChildPath ([System.Guid]::NewGuid().ToString())
-    $i++
-} while ((Test-Path -Path $tempPath) -and ($i -lt 100))
+$url            = "https://download.virtualbox.org/virtualbox/$packageVersion/$file"
+$checksum       = 'c987cdc8c08c579f56d921c85269aeeac3faf636babd01d9461ce579c9362cdd'
+$checksumType   = 'sha256'
+$unzipLocation  = "$ENV:TMP\$file"
 
-# check we've not hit the count limit
-if ($i -ge 100) {
-  Write-Error "We couldn't find a temp folder name to create! (count: $i)"
-}
 
-# Create the folder
-$null = New-Item -ItemType Directory -Path $tempPath
+# Download + extract the ISO
+	$zipArgs = @{
+		packageName   = $ENV:ChocolateyPackageName
+		url           = $url
+		unzipLocation = $unzipLocation
+		checksum      = $checksum
+		checksumType  = $checksumType
+	}
 
-$packageArgs = @{
-  packageName    = $env:ChocolateyPackageName
-  fileFullPath   = Join-Path -Path $toolsDir -ChildPath 'VBoxGuestAdditions.iso'
-  destination    = Join-Path -Path $tempPath -ChildPath 'extracted'
-  fileType       = 'ZIP'
-}
+	Install-ChocolateyZipPackage @zipArgs
 
-Get-ChocolateyUnzip @packageArgs
 
-$binX86Filename = 'VBoxWindowsAdditions-x86.exe'
-$binX64Filename = 'VBoxWindowsAdditions-amd64.exe'
-$installArgs = @{
-  packageName = $env:ChocolateyPackageName
-  fileType = 'EXE'
-  file = Join-Path -Path $packageArgs.destination -ChildPath $binX86Filename
-  file64 = Join-Path -Path $packageArgs.destination -ChildPath $binX64Filename
-  silentArgs = '/S'
-  validExitCodes = @(0)
-}
-
-# certificate can be found inside the GPL version of virtualbox:
-# https://www.virtualbox.org/svn/vbox/trunk/src/VBox/HostDrivers/Support/Certificates/Trusted-OracleCorporationVirtualBox-05308b76ac2e15b29720fb4395f65f38.cer
-# current is expired 3/23/2022
-$certFile = Join-Path -Path $toolsDir -ChildPath "Trusted-OracleCorporationVirtualBox-05308b76ac2e15b29720fb4395f65f38.cer"
-certutil -addstore -f "TrustedPublisher" $certFile
-
-Install-ChocolateyInstallPackage @installArgs
+# Download + install the Oracle certificate
+	$certFileName = 'Trusted-OracleCorporationVirtualBox-05308b76ac2e15b29720fb4395f65f38.cer'
+	$certUrl      = "https://www.virtualbox.org/svn/vbox/trunk/src/VBox/HostDrivers/Support/Certificates/$certFileName"
+	$certFile     = Join-Path $unzipLocation $certFileName
+	$WebClient    = [System.Net.WebClient]::new()
+	$WebClient.DownloadFile($certUrl, $certFile)
+
+	$certFile       = Join-Path $unzipLocation "Trusted-OracleCorporationVirtualBox-05308b76ac2e15b29720fb4395f65f38.cer"
+	certutil -addstore -f "TrustedPublisher" $certFile
+
+
+# Install the EXE
+	$packageArgs = @{
+		packageName    = $ENV:ChocolateyPackageName
+		fileType       = 'EXE'
+		file           = Join-Path $unzipLocation $binX86Filename
+		file64         = Join-Path $unzipLocation $binX64Filename

```

Comment:
You haven't added the checksums for the binaries. These are also needed for the Chocolatey Community Repository.